examples/angular/src/app/lib/resolver/dotcms-page.resolver.ts

```diff
@@ -0,0 +1,59 @@
+import { inject } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  ResolveFn,
+  RouterStateSnapshot,
+} from '@angular/router';
+import { Observable, from, map, tap } from 'rxjs';
+
+import { DOTCMS_CLIENT_TOKEN } from '../dotcms-client-token';
+import { DotCMSNavigationItem, DotCMSPageAsset } from '../models';
+import { PageContextService } from '../services/dotcms-context/page-context.service';
+
+/**
+ * This resolver is used to fetch the page and navigation data from dotCMS.
+ *
+ * @param {ActivatedRouteSnapshot} route
+ * @param {RouterStateSnapshot} _state
+ * @return {*}
+ */
+export const DotCMSPageResolver: ResolveFn<
+  Observable<{
+    pageAsset: DotCMSPageAsset;
+    nav: DotCMSNavigationItem;
+  }>
+> = (route: ActivatedRouteSnapshot) => {
+  // Get the Service
+  const client = inject(DOTCMS_CLIENT_TOKEN);
```

Comment:
I think, we need create a service wrapper of the client, and the TOKEN need to be the config, no the client itself.

Reply:
We discuss and agree we this approach @oidacra. Can we talk about when @fmontes and @KevinDavilaDotCMS come back?